src/scarabee/_scarabee/cmfd.cpp

```diff
@@ -134,8 +130,32 @@ CMFD::CMFD(const std::vector<double>& dx, const std::vector<double>& dy,
   xs_.resize({nx_, ny_});
   xs_.fill(nullptr);
 
-  // Allocate the flux array
+  // Allocate cell volume array
+  volumes_.resize(nx_ * ny_);
+  for (std::size_t i = 0; i < nx_; i++) {
+    for (std::size_t j = 0; j < ny_; j++) {
+      // Store CMFD cell volume
+      const auto indx = tile_to_indx(i, j);
+      volumes_[indx] = dx_[i] * dy_[j];
+    }
+  }
+
+  // Set CMFD fluxes to 1
+  flux_cmfd_.resize(ng_ * nx_ * ny_);
+  flux_cmfd_.setOnes();
+
+  // Allocate flux update ratio array
+  update_ratios_.resize(ng_ * nx_ * ny_);
+  update_ratios_.setZero();
```

Comment:
I would initialize this to ones, just so we don't end up with a zero flux everywhere on accident one day.